src/server/ndjsonWorker.js

```diff
@@ -53,50 +53,50 @@ ndjsonWorker.process(async job => {
     return;
   }
 
-  const ndjsonLines = ndjsonResources.trim().split(/\n/);
-  if (ndjsonLines.length > 0 && JSON.parse(ndjsonLines[0]).resourceType === 'Parameters') {
-    // check first line for a Parameters header and remove if necessary
-    ndjsonLines.shift();
-  }
-  const insertions = ndjsonLines.map(async resourceStr => {
-    let data;
-    try {
-      data = JSON.parse(resourceStr);
-    } catch (e) {
-      throw new Error(`Error parsing JSON: ${resourceStr}`);
+  const ndjsonLines = ndjsonResources.split(/\n/);
+
+  const insertions = ndjsonLines.map(async (resourceStr, index) => {
+    resourceStr = resourceStr.trim();
+
+    // if line is empty skip
+    if (resourceStr === '') {
+      return null;
     }
+
+    // attempt to parse the line
     try {
+      const data = JSON.parse(resourceStr);
+
+      // check if first line is Parameters header and skip it
+      if (index === 0 && data.resourceType === 'Parameters') {
```

Comment:
The reason we don't want the initial `trim()` of the whole file is that it would screw up the line numbers in the output for where the parsing errors are. I'm reworking this to only skip parameters on the first non empty line.